secure/storage/src/vault.rs

```diff
@@ -127,6 +130,14 @@ impl VaultStorage {
         Ok(self.client.revoke_token_self()?)
     }
 
+    #[cfg(any(test, feature = "testing"))]
+    pub fn get_all_key_versions(
+        &self,
+        name: &str,
+    ) -> Result<Vec<ReadResponse<Ed25519PublicKey>>, Error> {
+        Ok(self.client().read_ed25519_key(name)?)
+    }
+
```

Comment:
just hold on to the first key, rotate 5 times, and ensure you cannot get the key.